python/src/aac/validate/_validate.py

```diff
@@ -56,7 +56,7 @@ def _validate_definitions(user_definitions: list[Definition]) -> ValidationResul
     def validate_each_definition(definition: Definition) -> list[ValidatorResult]:
         return _validate_definition(definition, registered_validators, active_context)
 
-    validator_results = list(flatten(map(validate_each_definition, user_definitions)))
+    validator_results = list(flatten(map(validate_each_definition, active_context.definitions)))
```

Comment:
I think I may be missing something here. Was this just a hole in a previous test that's being fixed?

Reply:
So, I had a question that never got asked about whether or not we should reconsider how we do validation. The question was whether we should carry on the old pattern (that's what this change does) or use a more directed pattern like the one in main that only validates the target validations. 

Why does this matter? We'll have to consider the impact of validating every definition in the context when we consider what APIs we're leveraging in the LSP diagnostic feature -- which is driven by our validation system.

I haven't taken the time to really think about the problem so I'm reverting to old validator behavior until we elect to consciously change the user's view of validation behavior. That being said, we were still testing the core spec in the workflow since we run the validate command on the core spec.

Reply:
Can we move this to a discussion, then? I thought we were still using the older pattern but we should probably discuss the pros and cons of each approach to make sure we're not missing anything.